modules/addon-irsa/iam-irsa.tf

```
@@ -18,13 +18,22 @@ data "aws_iam_policy_document" "irsa_assume" {
   }
 }
 
+data "aws_iam_policy_document" "irsa_policy" {
+  count = local.irsa_role_create && (var.irsa_policy_enabled || var.irsa_assume_role_enabled) ? 1 : 0
+
+  source_policy_documents = compact([
+    var.irsa_policy,
+    data.aws_iam_policy_document.irsa_assume[0].json
```

```suggestion
    one(data.aws_iam_policy_document.irsa_assume[*].json)
```
It might not be present when var. irsa_assume_role_enabled=false

---

I had it there and deleted , did not think about that use case :(